test/extensions/filters/http/cache/cache_filter_test.cc

```diff
@@ -114,19 +131,85 @@ class CacheFilterTest : public ::testing::Test {
     ::testing::Mock::VerifyAndClearExpectations(&decoder_callbacks_);
   }
 
+  // This function assumes that there is a body created using std::string(body_size, 'a').
+  void testSuccessfulValidation(CacheFilterSharedPtr filter, uint64_t body_size) {
+    ASSERT(body_size > 0);
+
+    // Make request require validation
+    request_headers_.setReferenceKey(Http::CustomHeaders::get().CacheControl, "no-cache");
+
+    // Decoding the request should find a cached response that requires validation.
+    // As far as decoding the request is concerned, this is the same as a cache miss with the
+    // exception of injecting validation precondition headers.
+    testDecodeRequestMiss(filter);
+
+    // Make sure validation conditional headers are added
+    const Http::TestRequestHeaderMapImpl injected_headers = {
+        {"if-none-match", etag_}, {"if-modified-since", response_last_modified_}};
```

Comment:
ditto

Reply:
To keep it simple and verbose, is that okay to keep it as is rather than using values in headers.h? The dic needs std::string pairs, so we will need to use something like Http::CustomHeaders.get().ifNoneMatch.get(), which is not clean. 

Besides, I checked other places where TestRequestHeaderMapImpl, it seems we generally follow a explicit key-value pair setting. WDYT?

Reply:
I'm OK either way. One way to make this a little less verbose than what you suggest is to make a temp for `Http::CustomHeaders.get()` and use that throughout this function. But it's up to you.